tests/countries/test_micronesia.py

```diff
@@ -0,0 +1,441 @@
+        )
+        self.assertHolidayName(f"{name} (observed)", dt)
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_independence_day(self):
+        name = "Independence Day"
+        self.assertHolidayName(name, (f"{year}-11-03" for year in range(1987, 2050)))
+        dt = (
+            "2001-11-02",
+            "2002-11-04",
+            "2007-11-02",
+            "2012-11-02",
+            "2013-11-04",
+            "2018-11-02",
+            "2019-11-04",
+            "2024-11-04",
+        )
+        self.assertHolidayName(f"{name} (observed)", dt)
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_fsm_veterans_of_foreign_wars_day(self):
+        name = "FSM Veterans of Foreign Wars Day"
+        self.assertHolidayName(name, (f"{year}-11-11" for year in range(2004, 2050)))
+        dt = (
+            "2006-11-10",
+            "2007-11-12",
+            "2012-11-12",
+            "2017-11-10",
+            "2018-11-12",
+            "2023-11-10",
+        )
+        self.assertHolidayName(f"{name} (observed)", dt)
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_presidents_day(self):
+        name = "Presidents Day"
+        self.assertHolidayName(name, (f"{year}-11-23" for year in range(2021, 2050)))
+        dt = (
+            "2024-11-22",
+            "2025-11-24",
+        )
+        self.assertHolidayName(f"{name} (observed)", dt)
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_christmas_day(self):
+        name = "Christmas Day"
+        self.assertHolidayName(name, (f"{year}-12-25" for year in range(1987, 2050)))
+        dt = (
+            "2004-12-24",
+            "2005-12-26",
+            "2010-12-24",
+            "2011-12-26",
+            "2016-12-26",
+            "2021-12-24",
+            "2022-12-26",
+        )
+        self.assertHolidayName(f"{name} (observed)", dt)
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_trk_holidays(self):
+        trk_holidays = Micronesia(subdiv="TRK", years=range(1987, 2050))
+
+        name = "State Charter Day"
+        self.assertHolidayName(name, trk_holidays, (f"{year}-09-26" for year in range(1987, 2050)))
+        dt = (
+            "2004-09-27",
+            "2009-09-25",
+            "2010-09-27",
+            "2015-09-25",
+            "2020-09-25",
+            "2021-09-27",
+        )
+        self.assertHolidayName(f"{name} (observed)", trk_holidays, dt)
+        self.assertNoNonObservedHoliday(dt)
+
+        name = "Chuuk State Constitution Day"
+        self.assertHolidayName(name, trk_holidays, (f"{year}-10-01" for year in range(1990, 2050)))
+        dt = (
+            "2000-10-02",
+            "2005-09-30",
+            "2006-10-02",
+            "2011-09-30",
+            "2016-09-30",
+            "2017-10-02",
+            "2022-09-30",
+            "2023-10-02",
+        )
+        self.assertHolidayName(f"{name} (observed)", trk_holidays, dt)
+        self.assertNoNonObservedHoliday(dt)
```

Comment:
We can directly replace `trk_holidays` with `self.subdiv_holidays["TRK"]`, same goes for other entries

Reply:
Look at Andorra or Argentina, for example.